pkg/chain/bitcoin/electrs.go

```diff
@@ -89,3 +89,34 @@ func (e ElectrsConnection) VbyteFee() (int32, error) {
 	logger.Info("retrieved a vbyte fee of [%d]", fee)
 	return int32(fee), nil
 }
+
+func (e ElectrsConnection) IsAddressUnused(btcAddress string) (bool, error) {
+	if e.apiURL == "" {
+		return true, nil
+	}
```

Comment:
This starts getting hairy. Returning `true` here may give us a false-positive result. 
Maybe we shouldn't have this `e.apiURL == ""` path included in all `electrs.go` functions and let it return an error if the connection details are not configured. In such case, we would need to decide in the code calling electrs function if it has the required details to call electrs or not.

Thoughts?

Reply:
the `e.apiURL == ""` is essentially a check to see if the user has explicitly turned off the electrs connection, which is we also have
```
func (e ElectrsConnection) Broadcast(transaction string) error {
	if e.apiURL == "" {
		for i := 0; i < 5; i++ {
			logger.Warningf("Please broadcast Bitcoin transaction %s", transaction)
		}
		return nil
	}
```
and 
```
// VbyteFee retrieves the 25-block estimate fee per vbyte on the bitcoin network.
func (e ElectrsConnection) VbyteFee() (int32, error) {
	if e.apiURL == "" {
		return 0, nil
	}
```

If there is no electrs connection, broadcast the transaction to the warning logs, and return 0 as the vbyte fee.

If there is no electrs connection, then ultimately the client has no way to check whether or not a btc address is unused - they can only rely on the next generated address from storage.

I'm down to re-write how that behavior is expressed! We can either put the defaulting behavior in the electrs code (like how we have it here), or we can move that up a level to the code that calls it. 

Currently in client.go, we have:
```
electrsConnection := bitcoin.NewElectrsConnection(tbtcConfig.Bitcoin.ElectrsURLWithDefault())
...
beneficiaryAddress, err := recovery.ResolveAddress(
  tbtcConfig.Bitcoin.BeneficiaryAddress,
  derivationIndexStorage,
  chainParams,
  electrsConnection,
)
...
vbyteFee, vbyteFeeError := electrsConnection.VbyteFee()
...
broadcastError := electrsConnection.Broadcast(recoveryTransactionHex)
```

We could rewrite that to check the `ElectrsURLWithDefault()` and see if it's `""`, and then make the default decision in the client code, but I'm not sure that's cleaner.

Reply:
I think we should return this status as an error, and match the bool return of any other error state. The expected behavior from the caller is the same: we can't determine whether the address is used or not with this method, use whatever backup approach you want.

I think most likely the correct bool return for an error state is `false`, simply because that's the zero value of `bool`.

Reply:
With that said, I _think_ most of these functions all have the same pattern, and I wonder if we can just abstract that pattern to a single central function. The pattern I'm seeing is:
- If `apiURL` is blank, return error.
- GET a particular address (= one parameter).
- If the status code is non-200, try to extract an error and return it as possible.
- If the status code is 200, JSON-parse.
- Do something with the JSON.

If I'm right about that, it means all but the “do something with the JSON” should be movable to a single function called by each of these, which prevents us from having this weird blank apiURL check in every function.

Reply:
> If apiURL is blank, return error.

None of them currently do this - if the apiURL is blank, they perform the no-electrs behavior. 0 fee for vbyte, send the transactions to the logger for broadcast, return true for validation, all with `nil` for the error.

I can definitely change this across the board though (by returning the nil value and a non-nil error)

Reply:
I think changing across the board makes sense. The contract with `bitcoin.Handle` is then “if there is an API response available, I will return it; if not, you will get an error”, and it's the caller's responsibility to default or handle the error some other way.